### PR TITLE
Make a pass over parametric graph section..

### PR DESCRIPTION
**Pull request by rocky**
Some type-checking corrections were made and some bugs found from this fixed.

### CODE REVIEW

File: pymathics/graph/base.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def graph_helper(
     root: Optional[int] = None,
     *args,
     **kwargs,
-) -> Optional[Callable]:
+) -> Optional["Graph"]:
```
Comment (mmatera) [inline pymathics/graph/base.py:89]: @rocky, if you are in this, maybe a docstring here would be useful too.

File: pymathics/graph/eval/parametric.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ def eval_full_rary_tree(
     self, r: Integer, n: Integer, expression, evaluation: Evaluation, options: dict
 ) -> Optional[Graph]:
     """
-    Call networkx to get a full_raray_tree using parameters, ``r`` and ``t``.
+    Call networkx to get a full_rary_tree using parameters, ``r`` and ``t``.
```
Comment (mmatera) [inline pymathics/graph/eval/parametric.py:39]: "full r-ary tree" would be easier to understand. 'rary' sound to me like "weirdy"

File: pymathics/graph/properties.py [hunk 3]
```diff
@@ -19,12 +19,11 @@ class AcyclicGraphQ(_NetworkXBuiltin):
     """
     <url>
     :Acyclic graph:
-    https://en.wikipedia.org/wiki/Acyclic_graph</url> (<url>
+    https://en.wikipedia.org/wiki/Acyclic_graph</url> test (<url>
```
Comment (mmatera) [inline pymathics/graph/properties.py:22]: `test` is for debug?

Comment (rocky) [reply pymathics/graph/properties.py:22]: No - test is what this function does this is an Acyclic graph _test_. This is not an Acyclic graph.

File: pymathics/graph/eval/parametric.py (hunk 2 above)
Comment (rocky) [reply pymathics/graph/eval/parametric.py:39]: networkx `full_rary_tree` is NetworkX's name. However I will revise this. I agree that something like full r-ary tree would be be better.

File: pymathics/graph/base.py (hunk 1 above)
Comment (rocky) [reply pymathics/graph/base.py:89]: I do not have a details idea of what graph_heper does and this will probably change. That said, I'll add something minimal and then we can fill it out later.

File: pymathics/graph/eval/parametric.py (hunk 2 above)
Comment (rocky) [reply pymathics/graph/eval/parametric.py:39]: Change that - lookin at this this is  pretty thin wrapper around  NetworkX's `full_rary_tree`. So I think best here to just make it clear that this is what it is doing. And you you want more detail about `full_rary_tree`  looking at the docs in networkx would be better.

File: pymathics/graph/properties.py (hunk 3 above)
Comment (mmatera) [reply pymathics/graph/properties.py:22]: Ah, I see. Thanks.

Comment (rocky) [reply pymathics/graph/properties.py:22]: You would do me a favor when you write documentation to actually _look_ at it in Django or in the LaTeX PDF.  There were many formatting and flaws that I think are easily seen when _looking_ at some form of formatted output.

Comment (mmatera) [reply pymathics/graph/properties.py:22]: Sure, but I did it over some cases. The problem was the number of new cases was very big, and I could not check all the cases. In any case, if there are general format guidelines, I could reinforce them in the documentation tests.

File: pymathics/graph/eval/parametric.py (hunk 2 above)
Comment (mmatera) [reply pymathics/graph/eval/parametric.py]: Thanks. And OK, but in any case, there is a link to the networkx documentation.

File: pymathics/graph/base.py (hunk 1 above)
Comment (mmatera) [reply pymathics/graph/base.py:89]: Thanks! Makes sense.

**Review by mmatera — APPROVED** [review]
LGTM

File: pymathics/graph/properties.py (hunk 3 above)
Comment (rocky) [reply pymathics/graph/properties.py:22]: I think our styles are different. You seem to prefer to make massive amounts of changes which have problems, and I prefer making small changes that are better quality, and _then_ expanding to everything. 

So from my standpoint if you aren't going to look at the documentation to see that it is right, please just don't "fix" this. Or do smaller changes which are easier to review. 

The same thing tends to be true for the stuff that is is enforcing "guidelines" it too often is done without thinking deeply about the problem or checking existing conventions but instead just plows ahead with its own (which sometimes can be buggy when combined with not looking) or is put in the wrong place such every test rather than in a git hook test or a workflow test.  So again, from my view it is better not to have these than put in  less-great quality checks. A problem with this code has been precisely the not great quality that we constantly are chasing around to try to fix.

**Comment by rocky** [comment]
Plenty more that remains, but this is getting big.

File: pymathics/graph/properties.py (hunk 3 above)
Comment (mmatera) [reply pymathics/graph/properties.py:22]: @rocky, I can not say if the documentation is "right". The only thing that I say is that it must be as uniform as possible. This is where "massivity" is important: If I want to have uniformity, the best is to do a lot at the time, to see the global picture. 
So, what I can do is mechanically complete a lot of URLs, add indentation, and follow your guidelines to improve the formatting. If I find a pattern, then I try to improve the quality/style tests in order to reinforce the pattern over all the documentation. Notice also that I never merge anything without waiting for the revision of you and @TiagoCavalcante. 

If this is not useful, of course, I can look at other aspects of the code, but I thought that this burst was needed in order to be close to the release.

In any case, now the documentation of all the elements in the module is "complete" in the sense that each symbol has the same minimal basic features (<dl>..</dl> block, external URLs, summary_text, and working doctests). This can be improved a lot, by adding more examples, fixing possible typos, and, as I did recently, and motivated by this task, improve a little bit the code of the doc generator in order to allow more readable sources.

Regarding the quality of the testing code, the only thing that I can promise is that I will try to avoid entangling it, so if eventually, someone wants to improve the quality of a piece of code, it can be done without having to go across all the project. Also, the consistency tests can be easily removed if you think that does not help in the development without too much effort.

Comment (rocky) [reply pymathics/graph/properties.py:22]: First and foremost - thanks for all of the hard work. Mathics3 wouldn't be nearly where it is today without all of your help, effort and good work. 

The challenge has been about focusing the work on the places where the effort is best put to use, and organizing work so that overall we are pushing in the same direction and ideally reinforcing ideas and efforts.

> So, what I can do is mechanically complete a lot of URLs, add indentation, and follow your guidelines to improve the formatting. 

Here is a suggestion. If you want to mechanize, just do a _small_ sample first and stop. Look at it and try it out and then get it reviewed. 

And then after making sure it doesn't have misfeature our outright bugs (like broken links and unintended line breaks), _then_ to larger (but not massively large) set. Let's see how that goes, because there will be more kinds of things to look for. For example maybe there are built-ins that don't have WMA links. Or maybe entries that don't quite have the right Wikipedia entries. That sort of thing. But rather than blindly doing something in these cases just stop.

A lot of this has to do with knowing one' strengths and one's limitations. And learning from past mistakes or adapting.   

Basically when you say want to do largish refactoring. That generally takes a lot of iteration and involvement of others. 

On the other hand finding and fixing WMA compliance is clearly you are well suited and good at. 

We are now I hope about week or two out from release. Making a pass over pymathics.natlang to get it closer to standards, getting out the last mathics-omnibus image and a final LaTeX doc fixes are the only things left.

Here is something that would be very very useful. Yes, we can build to docs, without errors so the PDF generation completes. 

But we get a _lot_ of warnings like missing cross reference or missing links and a number of other things like that.

Whatever you can do to reduce these would be most appreciated. 

And thanks for all the hard work.

File: pymathics/graph/eval/parametric.py (hunk 2 above)
Comment (rocky) [reply pymathics/graph/eval/parametric.py]: I think we are on the same page, but just to make sure...

If a number of IDE's and editors that support LSP (language server protocol), when you hover over "full_rary_tree"  in the code it will take you to the definition  or show you the docstring.
